### PR TITLE
Bluetooth: tester: Add provisioning procedure, Config Client and Health Client tests support

### PR DESCRIPTION
**Pull request by agata-ponitka**

### CODE REVIEW

File: subsys/bluetooth/mesh/cfg_cli.c [hunk 1]
```diff
@@ -115,6 +115,52 @@ static void gatt_proxy_status(struct bt_mesh_model *model,
...
+				       (void **)&param)) {
+		return;
+	}
+
+	status = net_buf_simple_pull_u8(buf);
+	net_idx = net_buf_simple_pull_le16(buf) & 0xfff;
+	phase = net_buf_simple_pull_u8(buf);
+
+	if (param->net_idx != net_idx) {
+		BT_WARN("Net Key Status key index does not match");
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/cfg_cli.c:152]: I believe this practice of warning about non-matching has stopped, as it just creates noise

File: subsys/bluetooth/mesh/cfg_cli.c [hunk 2]
```diff
@@ -115,6 +115,52 @@ static void gatt_proxy_status(struct bt_mesh_model *model,
...
+		return;
+	}
+
+	if (param->status) {
+		*param->status = status;
+		*param->phase = phase;
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/cfg_cli.c:158]: Check `param->phase == NULL` as well.
Same in the other two-parameter handlers

File: subsys/bluetooth/mesh/cfg_cli.c [hunk 3]
```diff
@@ -2021,6 +2406,123 @@ int bt_mesh_cfg_hb_pub_get(uint16_t net_idx, uint16_t addr,
...
+	}
+
+	return bt_mesh_msg_ack_ctx_wait(&cli->ack_ctx, K_MSEC(msg_timeout));
+}
+
+int bt_mesh_cfg_node_identity_get(uint16_t net_idx, uint16_t addr,
+				  uint16_t key_net_idx, uint8_t *status,
+				  uint8_t *identity)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, OP_NODE_IDENTITY_GET, 3);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/cfg_cli.c:2455]: Could be 2

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Comment by LingaoM** [comment]
@agata-ponitka Can you choose time to continue to promote this PR?

**Review by LingaoM — APPROVED** [review]

File: include/bluetooth/mesh/cfg_cli.h [hunk 4]
```diff
@@ -926,6 +954,109 @@ int bt_mesh_cfg_hb_pub_set(uint16_t net_idx, uint16_t addr,
...
+ *  @param addr      Target node address.
+ *  @param elem_addr Element address the model is in.
+ *  @param mod_id    Model ID.
+ *  @param pub       Publication parameters.
+ *  @param status    Status response parameter.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_mod_pub_va_set(uint16_t net_idx, uint16_t addr, uint16_t elem_addr, uint16_t mod_id,
+			       struct bt_mesh_cfg_mod_pub *pub, uint8_t *status);
```
[the remark this answers is not in the document]
Comment (trond-snekvik) [reply include/bluetooth/mesh/cfg_cli.h:1058]: Needs a vendor model variant as well

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
A couple of minor comments and one unresolved comment from the previous round :)

File: subsys/bluetooth/mesh/cfg_cli.c (hunk 2 above)
Comment (trond-snekvik) [reply subsys/bluetooth/mesh/cfg_cli.c:160]: Still could have `NULL` pointer access here

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
You'll need to rebase on `main` to resolve the config client conflict. There's also an unresolved NULL pointer access above. 

Otherwise good :+1:

**Review by trond-snekvik — APPROVED** [review]

**Comment by cfriedt** [comment]
@trond-snekvik - are all outstanding issues resolved? Ready to be merged?

**Review by LingaoM — APPROVED** [review]

**Comment by trond-snekvik** [comment]
> @trond-snekvik - are all outstanding issues resolved? Ready to be merged?

Yep, good to go!

**Review by jhedberg — APPROVED** [review]

**Pull request merged**